File: map_ts/src/Path/path-schedule.ts
import type { PathId } from "./path-id.js";
import type { PathState } from "./path-state.js";
import type { PathDataState } from "./path-data-state.js";

export type PathStateTransition = {
  from: PathState;
  to: PathState;
};


export type PathDataTransition = {
  from: PathDataState;
  to: PathDataState;
};


export type PathSchedule = {
  stateTransition: PathStateTransition;

  dataTransition: PathDataTransition;

  currentPath: PathId;

  nextPath?: PathId;
  destination?: PathId;
};

File: map_python/src/Path/path_schedule.py
from dataclasses import dataclass

from .path_data_state import PathDataState
from .path_id import PathId
from .path_state import PathState


@dataclass(slots=True)
class PathStateTransition:
    source: PathState
    target: PathState


@dataclass(slots=True)
class PathDataTransition:
    source: PathDataState
    target: PathDataState


@dataclass(slots=True)
class PathSchedule:
    state_transition: PathStateTransition
    data_transition: PathDataTransition
    current_path: PathId
    next_path: PathId | None = None
    destination: PathId | None = None
